src/send_packets.c

```diff
@@ -496,6 +502,16 @@ send_packets(tcpreplay_t *ctx, pcap_t *pcap, int idx)
             break;
         }
 
+        /* stop sending based on the duration limit*/
+        if (end_us > 0) {
+            if (gettimeofday(&now, NULL) < 0)
+                errx(-1, "gettimeofday() failed: %s",  strerror(errno));
+            if (TIMEVAL_TO_MICROSEC(&now) > end_us) {
```

Comment:
`gettimeofday()` only fails if there is a parameter problem. No need to check for errors when passing `NULL`